migrate.py

```diff
@@ -361,6 +362,10 @@ def resolve_spec(spec, checkoutdir):
                         # clean out glob entry
                         spec[ns][coll][ptype].remove(entry)
 
+                # NOTE now that spec for plugins of ptype has been finalized, we can iterate again and add files for the dupe check
+                for entry in spec[ns][coll][ptype]:
```

Comment:
You can do this w/o a loop:
```suggestion
                files_to_collections[os.path.join(plugin_base, entry)].extend(spec[ns][coll][ptype])
```